draft-ietf-quic-qlog-main-schema.md

```diff
@@ -1137,18 +1143,76 @@ SimulationMarker = {
 ~~~
 {: #simulation-marker-def title="SimulationMarker definition"}
 
-# Event definition guidelines
+# Event extensibility {#event-extensibility}
+
+The main qlog schema defined by this document describes log and trace formats
+that have related events. A minimal set of common events are presented in
+{{events}}. It is expected that logging is extended with specific, per-protocol
+event definitions that specify the name (category + type) and data needed for
+each individual event. Examples include the QUIC event definitions {{QLOG-QUIC}}
+and HTTP/3 event definitions {{QLOG-H3}}.
+
+New event definitions SHOULD follow the guidance in this section, which provides
+consistency that makes it easier for qlog implementers to support a wide variety
+of protocols.
+
+## Extended Event Schema
+
+New event definitions SHOULD be part of an extension schema, using the
+annotations and concepts presented in this section.
+
+An extension schema defines one or more qlog event categories that can contain
+multiple event definitions. Each category MUST have a globally unique category
+identifier. Each event MUST only belong to a single category. Per
+{{{#name-field}}}, event names are the concatenation of category and type.
+Therefore, these requirements ensure there are no ambiguities when multiple
+extension schema are in use.
+
+Implementations that might record events from extension schemas SHOULD list all
+category identifiers in use. This is achieved by including the appropriate URI
+in the `additional_event_schemas` field of the QlogFile ({{qlog-file-schema}})
+or QlogFileSeq ({{qlog-file-seq-schema}}). The `additional_event_schema` is a
+hint to tools about the possible events that a qlog file might contain. The file
+may contain events that do not belong to a listed category identifier. Tools
+MUST NOT treat this as an error; see {{tooling}}.
+
+Each extension schema is named by a URI. That URI MUST be absolute and MUST
+include a category identifier, indicated using a fragment identifier (characters
+after a "#" in the URI).
+
+For extension schema defined in RFCs, the URI used SHOULD be a URN starting with
+`urn:ietf:params:qlog:` followed by a registered, descriptive name.
+
+Private or non-standard extension schema a can use other URI formats. URIs that
+contain a domain name SHOULD also contain a month-date in the form mmyyyy. The
+definition of the schema and assignment of the URI MUST have been authorized by
+the owner of the domain name on or very close to that date. (This avoids
+problems when domain names change ownership.)
+
+In the following example, a qlog file contains events defined in two hypothetical standardized
+extension schema, along with a private extension. The first schema is named
```

Comment:
it's a bit confusing that here we have 3 schema in a single file... while not impossible, most files will have a single schema (i.e., quic and h3).

I would rework this to saying that we have a single file with 2 schema (rick and john) and another file with a private extension schema (pickle).

Reply:
I don't think its confusing. For the HTTP/3 qlog I'd generate for quiche, I'm already expecting to do the following based on identifiers in the QUIC and H3 extension schema

```
~~~
"additional_event_schemas": [
                            "urn:ietf:params:qlog:quic#quic",
                            "urn:ietf:params:qlog:quic#security",
                            "urn:ietf:params:qlog:quic#recovery",
                            "urn:ietf:params:qlog:h3#h3"
                          ]
~~~